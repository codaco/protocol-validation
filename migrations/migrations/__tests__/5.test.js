/* eslint-env jest */

const version5 = require('../5');

const v4protocol = {
  codebook: {
    node: {
      exampleNodeType: {
        variables: {
          invalidExampleVariable: {
            name: 'variable (with) disallowed characters!',
            options: [
              { label: 'foo', value: 'f o o!' },
              { label: 'bar', value: 'b_a-r:.' },
              { label: 'bazz', value: 5 },
            ],
          },
          validExampleVariable: {
            name: 'variable_with-allowed:characters.',
          },
        },
      },
    },
  },
};

describe('migrate v4 -> v5', () => {
  it('variable names', () => {
    const result = version5.migration(v4protocol);

    const variables = result.codebook.node.exampleNodeType.variables;
    expect(variables.invalidExampleVariable.name).toBe('variable_with_disallowed_characters');
    expect(variables.validExampleVariable.name).toBe('variable_with-allowed:characters.');
    expect(result).toMatchSnapshot();
  });

  it('option values', () => {
    const result = version5.migration(v4protocol);

    const options = result.codebook.node.exampleNodeType.variables.invalidExampleVariable.options;
    expect(options).toEqual(expect.arrayContaining([
      expect.objectContaining({ value: 'f_o_o' }),
      expect.objectContaining({ value: 'b_a-r:.' }),
      expect.objectContaining({ value: 5 }),
    ]));
    expect(result).toMatchSnapshot();
  });
});
