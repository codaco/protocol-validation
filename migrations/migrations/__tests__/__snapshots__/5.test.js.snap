// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migrate v4 -> v5 option values 1`] = `
Object {
  "codebook": Object {
    "node": Object {
      "exampleNodeType": Object {
        "variables": Object {
          "invalidExampleVariable": Object {
            "name": "variable_with_disallowed_characters",
            "options": Array [
              Object {
                "label": "foo",
                "value": "f_o_o",
              },
              Object {
                "label": "bar",
                "value": "b_a-r:.",
              },
              Object {
                "label": "bazz",
                "value": 5,
              },
            ],
          },
          "validExampleVariable": Object {
            "name": "variable_with-allowed:characters.",
          },
        },
      },
    },
  },
}
`;

exports[`migrate v4 -> v5 variable names 1`] = `
Object {
  "codebook": Object {
    "node": Object {
      "exampleNodeType": Object {
        "variables": Object {
          "invalidExampleVariable": Object {
            "name": "variable_with_disallowed_characters",
            "options": Array [
              Object {
                "label": "foo",
                "value": "f_o_o",
              },
              Object {
                "label": "bar",
                "value": "b_a-r:.",
              },
              Object {
                "label": "bazz",
                "value": 5,
              },
            ],
          },
          "validExampleVariable": Object {
            "name": "variable_with-allowed:characters.",
          },
        },
      },
    },
  },
}
`;
